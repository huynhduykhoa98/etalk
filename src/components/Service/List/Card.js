/* eslint-disable linebreak-style */
import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';

class Card extends Component {
  render() {
    return (
      <li>
        <img src={this.props.img} alt={this.props.maintext} width={100} />
        <div>
          <h3>
            <FormattedMessage
              id="serviceHeader"
              values={{ order: `${this.props.value}` }}
            />
          </h3>
          <FormattedMessage
            id="serviceText"
            values={{
              order: `${this.props.value}`,
              p: (chunks) => <p>{chunks}</p>,
            }}
          />
        </div>
      </li>
    );
  }
}

export default Card;
