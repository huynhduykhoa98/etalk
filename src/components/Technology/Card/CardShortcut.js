/* eslint-disable linebreak-style */
import React, { Component } from 'react';

class CardShortcut extends Component {
  render() {
    return (
      <a href={`${this.props.href}`} title={`${this.props.title}`}>
        <img src={`${this.props.img}`} alt={`${this.props.title}`} />
        {`${this.props.title}`}
      </a>
    );
  }
}

export default CardShortcut;
