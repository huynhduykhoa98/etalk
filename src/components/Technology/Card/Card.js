/* eslint-disable linebreak-style */
/* eslint-disable class-methods-use-this */
import React, { Component } from 'react';
import CardShortcut from './CardShortcut';

class Card extends Component {
  render() {
    return (
      <div className="pic">
        <CardShortcut
          href="//www.w3.org/standards/webdesign/htmlcss"
          title="GSAP"
          img="https://spider.vn/images/icon_gsap.svg"
        />
        <CardShortcut
          href="//www.w3.org/standards/webdesign/htmlcss"
          title="HTML & CSS"
          img="https://spider.vn/images/icon_html.png"
        />

        <CardShortcut
          href="//www.w3.org/standards/webdesign/script"
          title="JavaScript"
          img="https://spider.vn/images/icon_js.png"
        />

        <CardShortcut
          href="//babeljs.io"
          title="Babel"
          img="https://spider.vn/images/icon_babel.svg"
        />

        <CardShortcut
          href="//pugjs.org"
          title="Pug"
          img="https://spider.vn/images/icon_pugjs.svg"
        />

        <CardShortcut
          href="//sass-lang.com"
          title="Sass"
          img="https://spider.vn/images/icon_sass.svg"
        />

        <CardShortcut
          href="//getbootstrap.com"
          title="Bootstrap"
          img="https://spider.vn/images/icon_bootstrap.png"
        />

        <CardShortcut
          href="//material.io"
          title="Material"
          img="https://spider.vn/images/icon_material.png"
        />

        <CardShortcut
          href="//jquery.com"
          title="jQuery"
          img="https://spider.vn/images/icon_jquery.svg"
        />

        <CardShortcut
          href="//nextjs.org"
          title="Next.js"
          img="https://spider.vn/images/icon_nextjs.png"
        />

        <CardShortcut
          href="//angular.io"
          title="Angular"
          img="https://spider.vn/images/icon_angular.png"
        />

        <CardShortcut
          href="//reactjs.org"
          title="React"
          img="https://spider.vn/images/icon_react.svg"
        />

        <CardShortcut
          href="//redux.js.org"
          title="Redux"
          img="https://spider.vn/images/icon_redux.svg"
        />

        <CardShortcut
          href="//graphql.org"
          title="GraphQL"
          img="https://spider.vn/images/icon_graphql.svg"
        />

        <CardShortcut
          href="//vuejs.org"
          title="Vue"
          img="https://spider.vn/images/icon_vue.png"
        />

        <CardShortcut
          href="//lodash.com"
          title="Lodash"
          img="https://spider.vn/images/icon_lodash.svg"
        />

        <CardShortcut
          href="//underscorejs.org"
          title="Underscore"
          img="https://spider.vn/images/icon_underscore.png"
        />

        <CardShortcut
          href="//backbonejs.org"
          title="Backbone.js"
          img="https://spider.vn/images/icon_backbone.png"
        />

        <CardShortcut
          href="//emberjs.com"
          title="Ember.js"
          img="https://spider.vn/images/icon_ember.png"
        />

        <CardShortcut
          href="//phaser.io"
          title="Phaser"
          img="https://spider.vn/images/icon_phaser.png"
        />

        <CardShortcut
          href="//threejs.org"
          title="Three.js"
          img="https://spider.vn/images/icon_threejs.png"
        />

        <CardShortcut
          href="//d3js.org"
          title="D3.js"
          img="https://spider.vn/images/icon_d3.png"
        />

        <CardShortcut
          href="//www.gatsbyjs.org"
          title="Gatsby"
          img="https://spider.vn/images/icon_gatsby.svg"
        />

        <CardShortcut
          href="//www.webcomponents.org"
          title="Web Components"
          img="https://spider.vn/images/icon_web_components.svg"
        />

        <CardShortcut
          href="//www.polymer-project.org"
          title="Polymer"
          img="https://spider.vn/images/icon_polymer.svg"
        />

        <CardShortcut
          href="//developers.google.com/web/progressive-web-apps"
          title="Progressive Web Apps"
          img="https://spider.vn/images/icon_web_app.png"
        />

        <CardShortcut
          href="//serverless.css-tricks.com"
          title="Serverless"
          img="https://spider.vn/images/icon_serverless.svg"
        />

        <CardShortcut
          href="//headlesscms.org/"
          title="Headless CMS"
          // img="https://spider.vn/images/icon_headlesscms.svg"
        />

        <CardShortcut
          href="//www.php.net"
          title="PHP"
          img="https://spider.vn/images/icon_php.svg"
        />

        <CardShortcut
          href="//laravel.com"
          title="Laravel"
          img="https://spider.vn/images/icon_laravel.svg"
        />

        <CardShortcut
          href="//octobercms.com"
          title="October"
          img="https://spider.vn/images/icon_october.png"
        />

        <CardShortcut
          href="//symfony.com"
          title="Symfony"
          img="https://spider.vn/images/icon_polymer.svg"
        />

        <CardShortcut
          href="//www.yiiframework.com"
          title="Yii"
          img="https://spider.vn/images/icon_yii.svg"
        />

        <CardShortcut
          href="//www.codeigniter.com"
          title="CodeIgniter"
          img="https://spider.vn/images/icon_codeigniter.svg"
        />

        <CardShortcut
          href="//wordpress.org"
          title="WordPress"
          img="https://spider.vn/images/icon_wordpress.svg"
        />

        <CardShortcut
          href="//woocommerce.com"
          title="WooCommerce"
          img="https://spider.vn/images/icon_woocommerce.svg"
        />

        <CardShortcut
          href="//www.drupal.org"
          title="Drupal"
          img="https://spider.vn/images/icon_drupal.svg"
        />

        <CardShortcut
          href="//golang.org"
          title="Golang"
          img="https://spider.vn/images/icon_golang.svg"
        />

        <CardShortcut
          href="//cordova.apache.org"
          title="Cordova"
          img="https://spider.vn/images/icon_cordova.svg"
        />

        <CardShortcut
          href="//ionicframework.com"
          title="Ionic"
          img="https://spider.vn/images/icon_ionic.svg"
        />

        <CardShortcut
          href="//facebook.github.io/react-native"
          title="React Native"
          img="https://spider.vn/images/icon_react.svg"
        />

        <CardShortcut
          href="//www.nativescript.org"
          title="NativeScript"
          img="https://spider.vn/images/icon_nativescript.svg"
        />

        <CardShortcut
          href="//flutter.dev"
          title="Flutter"
          img="https://spider.vn/images/icon_flutterio.svg"
        />

        <CardShortcut
          href="//electronjs.org"
          title="Electron"
          img="https://spider.vn/images/icon_electronjs.svg"
        />

        <CardShortcut
          href="//nodejs.org"
          title="Node.js"
          img="https://spider.vn/images/icon_nodejs.svg"
        />

        <CardShortcut
          href="//expressjs.com"
          title="Express"
          // img="https://spider.vn/images/expressjs.svg"
        />

        <CardShortcut
          href="//hapi.dev"
          title="Hapi"
          img="https://spider.vn/images/icon_hapi.svg"
        />

        <CardShortcut
          href="//sailsjs.com"
          title="Sails"
          img="https://spider.vn/images/icon_sails.svg"
        />

        <CardShortcut
          href="//sequelize.org"
          title="Sequelize"
          img="https://spider.vn/images/icon_sequelizejs.svg"
        />

        <CardShortcut
          href="//www.meteor.com"
          title="Meteor"
          img="https://spider.vn/images/icon_meteor.svg"
        />

        <CardShortcut
          href="//www.sqlite.org"
          title="SQLite"
          img="https://spider.vn/images/icon_sqlite.svg"
        />

        <CardShortcut
          href="//www.mysql.com"
          title="MySQL"
          img="https://spider.vn/images/icon_mysql.svg"
        />

        <CardShortcut
          href="//www.postgresql.org"
          title="PostgreSQL"
          img="https://spider.vn/images/icon_postgresql.svg"
        />

        <CardShortcut
          href="//www.mongodb.com"
          title="MongoDB"
          img="https://spider.vn/images/icon_mongodb.svg"
        />

        <CardShortcut
          href="//redis.io"
          title="Redis"
          img="https://spider.vn/images/icon_redis.svg"
        />

        <CardShortcut
          href="//www.npmjs.com"
          title="NPM"
          img="https://spider.vn/images/icon_npmjs.svg"
        />

        <CardShortcut
          href="//yarnpkg.com"
          title="Yarn"
          img="https://spider.vn/images/icon_yarn.svg"
        />

        <CardShortcut
          href="//gruntjs.com"
          title="Grunt"
          img="https://spider.vn/images/icon_gruntjs.svg"
        />

        <CardShortcut
          href="//gulpjs.com"
          title="Gulp"
          img="https://spider.vn/images/icon_gulp.svg"
        />

        <CardShortcut
          href="//parceljs.org"
          title="Parcel"
          img="https://spider.vn/images/icon_parcel.png"
        />

        <CardShortcut
          href="//webpack.js.org"
          title="Webpack"
          img="https://spider.vn/images/icon_webpack.svg"
        />

        <CardShortcut
          href="//eslint.org"
          title="ESLint"
          img="https://spider.vn/images/icon_eslint.svg"
          alt="ESLint"
        />

        <CardShortcut
          href="//istanbul.js.org"
          title="Istanbul"
          img="https://spider.vn/images/icon_istanbul.png"
        />

        <CardShortcut
          href="//prettier.io"
          title="Prettier"
          img="https://spider.vn/images/icon_prettier.png"
        />

        <CardShortcut
          href="//git-scm.com"
          title="Git"
          img="https://spider.vn/images/icon_git.svg"
        />

        <CardShortcut
          href="//www.docker.com"
          title="Docker"
          img="https://spider.vn/images/icon_docker.svg"
        />
      </div>
    );
  }
}

export default Card;
