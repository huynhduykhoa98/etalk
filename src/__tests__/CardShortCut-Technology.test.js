/* eslint-disable linebreak-style */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CardShortcut from '../components/Technology/Card/CardShortcut';
import Card from '../components/Technology/Card/Card';

Enzyme.configure({ adapter: new Adapter() });
it('test render', () => {
  const wrapped = shallow(<Card />);
  expect(wrapped.find(CardShortcut).length).toEqual(66);
});
