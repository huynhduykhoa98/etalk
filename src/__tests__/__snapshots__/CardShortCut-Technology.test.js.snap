// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  href="undefined"
  title="undefined"
>
  <img
    alt="undefined"
    src="undefined"
  />
  undefined
</a>
`;
